### PR TITLE
jsonnet workflow improvements

### PR DESCRIPTION
**Pull request by tpetr**
This PR does a few things:
- Updates the workflows Makefile to additionally depend on the libsonnet files, so that a change to a lib regenerates the yamls.
- Updates `ojsonnet` to only output one trailing newline (the double newlines was triggering the "fix newlines" pre-commit check which made committing workflow changes a pain). I did this by adding an optional `newline` arg to `pr` and `pr2`. LMK if folks think there's a better way to do this.
- Exempts the generated yamls from the prettier pre-commit check. IMO the ojsonnet output is readable enough and removing the friction of having to prettier all your changes is nice
- Regenerates the yamls in response to the changes above

test plan:
- tests pass
- structural diffs are same

### CODE REVIEW

File: tools/ojsonnet/Main.ml [hunk 1]
```diff
@@ -80,12 +80,12 @@ let run (conf : conf) =
   match conf.format with
   | JSON ->
       let str = JSON.string_of_json json in
-      pr str
+      pr ~newline:false str
```
Comment (aryx) [inline tools/ojsonnet/Main.ml]: I would keep pr/pr2 like they are, and just use print_string str directly

**Review by aryx — APPROVED** [review]
Thank you! Great improvements to the ergonomy of using jsonnet for GHA.
I was also annoyed by all of this friction.

**Comment by tpetr** [comment]
Not sure why `license/cla` isnt being reported, but I work for Semgrep so I'm overriding this

**Comment by aryx** [comment]
It's a bit sad to learn it from license/cla, but yes, you've been fired last week but nobody wanted to tell you.

**Comment by tpetr** [comment]
my-watch-has-ended

**Pull request merged**